R/derive_advs_params.R

```diff
@@ -0,0 +1,317 @@
+  filter <- assert_filter_cond(enexpr(filter), optional = TRUE)
+  get_unit_expr <- assert_expr(enexpr(get_unit_expr), optional = TRUE)
+
+  derive_param_ratio(
+    dataset,
+    filter = !!filter,
+    numerator_code = wstcir_code,
+    denominator_code = hipcir_code,
+    by_vars = by_vars,
+    set_values_to = set_values_to,
+    get_unit_expr = !!get_unit_expr
+  )
+}
+
+#' Adds a Parameter for Waist to Height Ratio
+#'
+#' @description Adds a record for Waist to Height Ratio using Waist Circumference and Height
+#' each by group (e.g., subject and visit) where the source parameters are available.
+#'
+#' **Note:** This is a wrapper function for the more generic
+#' \code{admiral::derive_param_computed()}.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the \code{by_vars} argument are expected to be in the dataset.
+#'   \code{PARAMCD}, and \code{AVAL} are expected as well.
+#'
+#'   The variable specified by \code{by_vars} and \code{PARAMCD} must be a unique key of
+#'   the input dataset after restricting it by the filter condition (\code{filter} argument)
+#'   and to the parameters specified by \code{wstcir_code} and \code{height_code}.
+#'
+#' @param wstcir_code Waist Circumference parameter code
+#'
+#'   The observations where \code{PARAMCD} equals the specified value are considered
+#'   as the Waist Circumference
+#'
+#'   *Permitted Values:* character value
+#'
+#' @param height_code Height parameter code
+#'
+#'   The observations where \code{PARAMCD} equals the specified value are considered
+#'   as the Height. It is expected that Height is measured in cm
+#'
+#'   *Permitted Values:* character value
+#'
+#' @param constant_by_vars By variables for when Height is constant
+#'
+#'   When Height is constant, the Height parameters (measured only once) are merged
+#'   to the other parameters using the specified variables.
+#'
+#'   If Height is constant (e.g. only measured once at screening or baseline) then use
+#'   \code{constant_by_vars} to select the subject-level variable to merge on (e.g. \code{USUBJID}).
+#'   This will produce Waist to Height Ratio at all visits where Waist Circumference is measured.
+#'   Otherwise it will only be calculated at visits with both Height and Waist Circumference
+#'   collected.
+#'
+#'   *Permitted Values*: list of variables created by \code{exprs()}
+#'   e.g. \code{exprs(USUBJID, VISIT)}
+#'
+#' @inheritParams derive_param_ratio
+#'
+#' @details
+#' The analysis value of the new parameter is derived as
+#' \deqn{WAISTHGT = \frac{WSTCIR}{HEIGHT}}{WAISTHGT = WSTCIR / HEIGHT}
+#'
+#'
+#' @return The input dataset with the new parameter added. Note, a variable will only
+#'    be populated in the new parameter rows if it is specified in \code{by_vars}.
+#'
+#' @family der_prm_advs
+#' @keywords der_prm_advs
+#'
+#' @export
+#'
+#' @seealso \code{\link[admiral:derive_param_computed]{admiral::derive_param_computed()}}
```

Comment:
add the other ratio function here too

Reply:
Good point. Thanks.

Reply:
All functions from the same family are actually added there automatically.

![image](https://github.com/user-attachments/assets/f5f25867-12fc-4fca-bbc9-cacb6023074a)